Fixed a freeze error while loading, default delay configuration changed, source auto-formatted

TonEnfer opened the pull request.

TonEnfer committed:
Fixed a freeze error while loading
Default delay configuration changed

TonEnfer committed:
Edit launch config for use my STLink

MikhailChe commented on Core/Src/apploader.c:
```diff
@@ -329,7 +329,7 @@ bool AppLoader_verify_firmware() {
 
 void AppLoader_load_application() {
-
+	__disable_irq();
```
Вот это странно, что оно работает. Когда я пробовал так фиксить, то для systick в основной прогармме не включались прерывания, таймер не увеличивался и всё фризилось.

TonEnfer replied:
Однако, оно работает.
По идее, в этот момент прерывания нам не нужны уже, потому что дальше только установка стекпоинтера и загрузка в аппликуху

TonEnfer replied:
На самом деле логично, что это ничего не сломало, потому что сейчас AppLoader_load_application() вызывается в самом начале, когда прерывания итак выключены. Т.е. можно не вызывать __disable_irq();

MikhailChe commented on Core/Src/apploader.c:
```diff
@@ -339,6 +339,7 @@ void AppLoader_load_application() {
 
 	void(*app)(void) = (void(*)(void)) app_jump_address;
 
+
```
Лишняя строчка

MikhailChe commented on Core/Src/config.c:
```diff
@@ -36,8 +36,8 @@ static void makeDefaultConfig() {
 				.wait_for_flash_activation_confirmation_ms = 3000,
-				.activate_flash_drive_timeout_ms = 3000,
-				.activate_main_program_timeout_ms = 3000,
+				.activate_flash_drive_timeout_ms = 0,
```
зачем так жестоко? Не хочется ждать?

TonEnfer replied:
ага

MikhailChe commented on Core/Src/main.c:
```diff
@@ -94,83 +94,100 @@ void printHex(const char *prefix, size_t length, uint8_t *buffer) {
+	while(i!=0)
+		i--;
+
+	__IO uint32_t *check_flag_address = (__IO uint32_t*) (0x38800000);
```
check_flag_address -> load_app_flag или bootloader_app_switch

MikhailChe commented on Core/Src/main.c:
```diff
@@ -94,83 +94,100 @@ void printHex(const char *prefix, size_t length, uint8_t *buffer) {
+
+	__IO uint32_t *check_flag_address = (__IO uint32_t*) (0x38800000);
+
+	if (*check_flag_address == 0xAA55BB66) {
```
Вынести 0xAA55BB66 в константы

MikhailChe commented on Core/Src/main.c:
```diff
@@ -94,83 +94,100 @@ void printHex(const char *prefix, size_t length, uint8_t *buffer) {
+
+
 	HAL_StatusTypeDef status = HAL_ERROR;
 	/*
```
давай всё закоментированное уберём здесь. Оно тут лишнее всё-равно

MikhailChe commented on Core/Src/main.c:
```diff
@@ -288,99 +305,98 @@ int main(void)
-		log_info("Firmware check ok: code 0x%02X", aplres);
-		log_info("Updating firmware");
+		log_info("New firmware check ok: code 0x%02X", aplres);
+		log_info("Updating firmware...");
```
`<sarcasm>Три точки это обязательно</sarcasm>`

MikhailChe commented on Core/Src/main.c:
```diff
@@ -288,99 +305,98 @@ int main(void)
-	log_info("Loading application");
-	AppLoader_load_application();
+	log_info("Loading application...");
+	HAL_Delay(1000);
```
А зачем здесь delay?

TonEnfer replied:
потому что без него надпись "Loading application..." не видно :)

MikhailChe replied:
Надо вернуть здесь delay из конфигурации. Отписался выше.

MikhailChe commented on Core/Src/main.c:
```diff
@@ -288,99 +305,98 @@ int main(void)
-		log_info("Updating firmware");
+		log_info("New firmware check ok: code 0x%02X", aplres);
+		log_info("Updating firmware...");
 		if (!AppLoader_update_firmware()) {
```
Если не смогли обновить прошивку, нужно отменять загрузку приложения и сбрасывать магический флаг.

MikhailChe commented on milk-dispenser-bootloader Debug.launch:
```diff
@@ -5,11 +5,11 @@
     <intAttribute key="com.st.stm32cube.ide.mcu.debug.launch.formatVersion" value="2"/>
     <stringAttribute key="com.st.stm32cube.ide.mcu.debug.launch.ip_address_local" value="localhost"/>
-    <stringAttribute key="com.st.stm32cube.ide.mcu.debug.launch.loadList" value="{&quot;fItems&quot;:[{&quot;fIsFromMainTab&quot;:false,&quot;fPath&quot;:&quot;Debug/milk-dispenser.elf&quot;,&quot;fProjectName&quot;:&quot;milk-dispenser&quot;,&quot;fBuildConfigId&quot;:&quot;com.st.stm32cube.ide.mcu.gnu.managedbuild.config.exe.debug.1555714417&quot;,&quot;fPerformBuild&quot;:false,&quot;fDownload ...
+    <stringAttribute key="com.st.stm32cube.ide.mcu.debug.launch.loadList" value="{&quot;fItems&quot;:[{&quot;fIsFromMainTab&quot;:false,&quot;fPath&quot;:&quot;Debug/milk-dispenser.elf&quot;,&quot;fPerformBuild&quot;:false,&quot;fDownload&quot;:false,&quot;fLoadSymbols&quot;:false},{&quot;fIsFromMainTab&quot;:true,&quot;fPath&quot;:&quot;Debug\\milk-dispenser-bootloader.elf&quot;,&quot;fProjectNa ...
```
Надо будет подумать о том чтобы добавить конфигурации Лонча в gitignore

TonEnfer committed:
NOT TESTED! Fixed premature loading of the application

Add Backup RAM as not cached region in MPU
Change MCU Revision in CubeMX project

TonEnfer committed:
Fixed premature loading of the application

Add Backup RAM as not cached region in MPU
Change MCU Revision in CubeMX project

TonEnfer committed:
Merge branch 'master' of github.com:TonEnfer/milk-dispenser-bootloader

MikhailChe commented on .mxproject:
```diff
@@ -9,50 +9,50 @@ CDefines=USE_HAL_DRIVER;STM32H743xx;USE_HAL_DRIVER;USE_HAL_DRIVER;
-HeaderFiles#13=C:/Users/Mikhail/STM32CubeIDE/workspace_1.5.0/milk-dispenser-bootloader/Core/Inc/stm32h7xx_it.h
-HeaderFiles#14=C:/Users/Mikhail/STM32CubeIDE/workspace_1.5.0/milk-dispenser-bootloader/Core/Inc/stm32h7xx_hal_conf.h
-HeaderFiles#15=C:/Users/Mikhail/STM32CubeIDE/workspace_1.5.0/milk-dispenser-bootloader/Core/Inc/main.h
+HeaderFiles#0=D:/Docs/Prog/SMT32CubeWorkspace/milk-dispenser-bootloader/Core/Inc/gpio.h
```
Этот файл тоже ignore надо добавить

MikhailChe commented on Core/Src/main.c:
```diff
@@ -91,86 +95,72 @@ void printHex(const char *prefix, size_t length, uint8_t *buffer) {
+	/* USER CODE BEGIN 1 */
 
-	FLASH->OPTCCR |= FLASH_OPTCR_MER;
+//	SCB->CACR |= 1<<2;
```
коммент

MikhailChe commented on Core/Src/config.c:
```diff
@@ -36,8 +36,8 @@ static void makeDefaultConfig() {
-				.activate_flash_drive_timeout_ms = 3000,
-				.activate_main_program_timeout_ms = 3000,
+				.activate_flash_drive_timeout_ms = 0,
+				.activate_main_program_timeout_ms = 0,
```
Чтобы было видно Loading application давай здесь вернём не ноль, а 1000. А то в том месте получается магическая константа
